### PR TITLE
Update the Cloud team goals, milestones, roadmap

### PR DESCRIPTION
**Pull request by christinaforney**
Review by commit:

1. Move goals onto separate page
1. Update goals to focus on private code
1. Move roadmap to goals page and update milestones

@tsenart @ryanslade @quinnkeast - I propose merging this PR as is to get these high-level changes in. Then Ryan and Quinn can create a new/cleaner PR for updating the roadmap. Let me know if you have any concerns!

@nicksnyder - FYI, we are actively working on improving this, and this will only be a short term state.

### CODE REVIEW

**Review by tsenart — APPROVED** [review]

File: handbook/engineering/cloud/goals.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+**Milestones:**
+
+1. ✅ The Sourcegraph organization and team members can add public code from GitHub.com, GitLab.com, and Bitbucket Cloud.
+1. The Sourcegraph organization and team members can add private code to Sourcegraph Cloud.
+1. 🔄 Any user can add public code from GitHub.com, GitLab.com, and Bitbucket Cloud.
+1. Organization admins can manage membership (wording WIP).
+1. Any user or organization can add private code to Sourcegraph Cloud for free before it's GA.
+1. Sourcegraph Cloud is Generally Available (GA).
```
Comment (nicksnyder) [inline handbook/engineering/cloud/goals.md:24]:
```suggestion
1. Sourcegraph Cloud is a paid product that is Generally Available (GA)
```

File: handbook/engineering/cloud/goals.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
+# Cloud goals and priorities
+
+## Guiding principles
+
+1. **Make it work**: Build the backend blocking work, and expose it (even in a roughly usable way) to the Sourcegraph team. We will be able to quickly surface any glaring issues and will have more thoughts around usability. Take shortcuts where possible (this is currently due to the team having fewer frontend resources).
+   - Where possible, making it work and usable should be combined (avoid duplicate efforts), but if it's possible to separate the two in order to move things forward, we should!
+1. **Make it usable**: The experience has been designed and thought through. We feel good about putting this in front of users, and they will find it valuable!
```
Comment (nicksnyder) [inline handbook/engineering/cloud/goals.md:7]: (I know this got moved but worth changing now)

I don't think "usable" is a high enough bar. Please change this to "polished" or something else more ambitious.

```suggestion
1. **Make it polished**: The experience has been designed and thought through. We feel good about putting this in front of users, and they will find it valuable!
```

File: handbook/engineering/cloud/goals.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
+# Cloud goals and priorities
+
+## Guiding principles
```
Comment (nicksnyder) [inline handbook/engineering/cloud/goals.md:3]: Any objection to me moving this up one level for all of engineering, not just cloud team?

File: handbook/engineering/cloud/goals.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+## Goals
+
+### Private code on Sourcegraph Cloud
+
+**Problem:** Customers who want to use Sourcegraph with their private code must setup and run a Sourcegraph instance on their own compute infrastructure. There are customers who want to use Sourcegraph but don't want to have to deploy and operate their own Sourcegraph instance and associated compute infrastructure. These customers expect to be able to search over only the code they care about, and are not inundated with public or irrelevant code. To host private code on Sourcegraph Cloud, we need to not only ensure the security of our product (which is important for on-premise deployments too), but we also have to ensure the security of our Sourcegraph Cloud infrastructure.
+
+**Milestones:**
+
+1. ✅ The Sourcegraph organization and team members can add public code from GitHub.com, GitLab.com, and Bitbucket Cloud.
+1. The Sourcegraph organization and team members can add private code to Sourcegraph Cloud.
```
Comment (nicksnyder) [inline handbook/engineering/cloud/goals.md:20]: Is the team going to be task switching from the next milestone (which is in progress) to this one (which is not yet started)? If not, should we move this after the next milestone which is already started?

**Review by nicksnyder — APPROVED** [review]

File: handbook/engineering/cloud/goals.md (hunk 3 above)
Comment (christinaforney) [reply handbook/engineering/cloud/goals.md:3]: No objection at all :)

File: handbook/engineering/cloud/goals.md (hunk 2 above)
Comment (christinaforney) [reply handbook/engineering/cloud/goals.md:7]: What we're trying to communicate here is a good user experience. Polished does not necessarily mean good UX and can also create the impression of over doing it. I also see how make it usable does not make a high enough bar. I'm curious if @quinnkeast has an idea about how to best capture this.

File: handbook/engineering/cloud/goals.md (hunk 4 above)
Comment (christinaforney) [reply handbook/engineering/cloud/goals.md:20]: We just swapped these two (I know that was hard to see with the move). I think the team will actually ave both these milestones in progress with the frontend work focused on the next item and the backend work focused on this one.

File: handbook/engineering/cloud/goals.md (hunk 2 above)
Comment (nicksnyder) [reply handbook/engineering/cloud/goals.md:7]: > Polished does not necessarily mean good UX

It does to me!

> can also create the impression of over doing it.

Yeah, don't want that.


What about "smooth"?

```suggestion
1. **Make it smooth**: The experience has been designed and thought through. We feel good about putting this in front of users, and they will find it valuable!
```

**Pull request merged**